### PR TITLE
added views

### PR DESCRIPTION
**Pull request by MikeHampf**
Quite a bit of the backed is working, but I seem to have created some redundant endpoints (I'm using assessment 3 as a template), and I need to add some functionality to alter data in the database i.e. generate put and delete requests.  The two 3rd party APIs will be accessed from the frontend based on data stored in the backend.

### CODE REVIEW

File: composer_listing_app/urls.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from django.urls import path
+from .views import All_listings, Listing
+
+urlpatterns = [
+    path("", All_listings.as_view(), name="all-listings"),
+    path("add/", Listing.as_view(), name="add_listing")
```
Comment (ljkarel11) [inline composer_listing_app/urls.py:6]: would suggest a consistent naming scheme for "name" (either hyphen or underscore separated)

File: composer_listing_app/views.py [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+
+    def get(self, request):
+        entries = Composer_listing_serializer(Composer_listing.objects.order_by('id'), many=True)
+        return Response(entries.data)
+    
+class Listing(APIView):
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
```
Comment (ljkarel11) [inline composer_listing_app/views.py:25]: Generally post requests should be under All_listings. The usual way this goes is All_listings will have a get and post method to be able to get all listings or add a new one, and Listing will have a get, put, and delete method to get, update, or delete a single listing.

File: list_app/views.py [hunk 3]
```diff
@@ -1,3 +1,37 @@
...
+class List_manager(APIView):
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        reply = {"listings": [], "total_listings": 0}
+        list = Listening_list.objects.get(listener_id=request.user.id)
+        print(list.id)
+        for listings in Composer_listing_item.objects.filter(listening_list_id=list.id):
+            reply["total_listings"]+=1
```
Comment (ljkarel11) [inline list_app/views.py:20]: Could just get the length of a list here using .length. However, I'd suggest you consider whether you even need a "total_listings" field, as you can always just get that amount by checking the length of "listings" in the response.

File: list_app/views.py [hunk 4]
```diff
@@ -1,3 +1,37 @@
...
+            entry = Composer_listing.objects.get(id=listings.composer_item_id)
+            print(entry.composer_name)
+            reply["listings"].append({"id":listings.id,
```
Comment (ljkarel11) [inline list_app/views.py:23]: I'd suggest offloading a lot of this object creation/management into a serializer.

File: list_app/views.py [hunk 5]
```diff
@@ -1,3 +1,37 @@
...
+                                          "entry": {
+                                              "work":entry.work_title,
+                                              "composer": entry.composer_name,
+                                              "date added": listings.date_added
+                                          }
+                                          })
+        return Response(reply, status=HTTP_200_OK)
+
```
Comment (ljkarel11) [inline list_app/views.py:31]: You'll probably want a method to post a new list item, as well as another view to get/put/delete a single list item.

**Review by ljkarel11 — COMMENTED** [review]
A good start, just a few improvements and additions to be done before you are done with views. Ensure you add \_\_pycache\_\_/ to your .gitignore so that it doesn't clutter your repo.

File: composer_listing_app/urls.py (hunk 1 above)
Comment (MikeHampf) [reply composer_listing_app/urls.py:6]: that's a typo.  I'll get it fixed right up.  thanks for catching that

File: composer_listing_app/views.py (hunk 2 above)
Comment (MikeHampf) [reply composer_listing_app/views.py:25]: I believe I've resolved this.  post is now in All_listings

File: list_app/views.py (hunk 3 above)
Comment (MikeHampf) [reply list_app/views.py:20]: Good point.  It's certainly not necessary, but I'm trying to find places where I can practice adding logic into my API calls

File: list_app/views.py (hunk 4 above)
Comment (MikeHampf) [reply list_app/views.py:23]: I'll work on that

File: list_app/views.py (hunk 5 above)
Comment (MikeHampf) [reply list_app/views.py:31]: Yes, I need to work out some more functionality with put and delete options

File: list_app/views.py (hunk 4 above)
Comment (MikeHampf) [reply list_app/views.py:23]: I'm not entirely sure how to implement this as a serializer, and I really don't want to break what I have working at the moment.  Maybe we could look at this as an aside?

**Pull request merged**